### DISCUSSION TITLE
need help with creating custom json from csv using particular fields from the source csv file

### DISCUSSION THREAD

**Discussion by oldbonez** [TomWright/dasel#197] [Q&A] [answered]
Hi all,

I need to create a custom json from csv by selecting particular fields from the csv source file. The source file has the following header format
`build;started;state;lines with code; url`
What I would like to achieve is to filter the builds that are in the state finished and the builds respective lines with code either in json array or hash.
Can you give me some pointers for this?
Thanks!

**Comment by oldbonez** [comment]
I got the most close with jq by using this
`cat dasel_test.csv| jq --slurp --raw-input --raw-output 'split("\n") | .[1:] | map(split(",")) |  map({"build": .[0],"state": .[2],"lines with code": .[3]})' | jq '.[] | select(.state == "Finished")'`
looking for the same thing with dasel

**Comment by TomWright** [comment, marked as answer]
Dasel currently only supports a comma delimited file but I'd be happy to add a flag to change that.

If this were comma delimited the following should work:
```bash
dasel -f my_file.csv -w json -m '.(state=finished)'
```
This will read my_file.csv into memory with `-f`, allow you to select multiple records with `-m`, output in JSON with `-w json` and filter only the finished rows with the selector.

Each line of output will be a json object.

Here's a runnable example: https://dasel.tomwright.me/s/892efbcb-8c30-4cd9-b65a-44ee892b9fa2

Sorry for the delayed response. I've been pretty busy recently and not had much time to catch up.

**Reply by oldbonez** [reply]
Thanks Tom!
The file delimiter is actually ','
The best I did on my own  was
`dasel -r csv -w json -f build_search.csv -m --format '{{ select ".build" }},{{ select ".lines with code" }}' '.[*]' --merge-input-documents`
Can you advise how a multiple selection statement can be combined with certain keys assigned to a particular value. I would like to have a JSON in the end only for the finished builds and their respective lines with code instead of the 5 keys overall

**Reply by TomWright** [reply]
I believe the following will do what you need!

```bash
dasel -f build_search.csv -w json -m '.(state=finished)' --format '{"build": {{ mustToJson .build }}, "code": {{ mustToJson (index $ "lines with code")}}}'
```

Note that
1. The `index $ "lines with code"` is the golang template way of accessing a variable/field that contains spaces.
2. This requires the development build of dasel as the `mustToJson` function used here is part of sprig and this new functionality hasn't been released yet.
